src/main/java/com/google/devtools/build/lib/skyframe/IgnoredPackagePrefixesFunction.java

```diff
@@ -142,5 +157,23 @@ public String extractTag(SkyKey skyKey) {
     public IgnoredPatternsFunctionException(InconsistentFilesystemException e) {
       super(e, Transience.TRANSIENT);
     }
+
+    public IgnoredPatternsFunctionException(RepositoryFetchingException e) {
+      super(e, Transience.TRANSIENT);
+    }
+  }
+
+  /**
+   * Repackages the Exceptions that could arise from fetching an external repository.
+   */
+  public static class RepositoryFetchingException extends Exception {
+    public RepositoryFetchingException(Exception e) {
+      super(e.getMessage(), e.getCause() == null ? e : e.getCause());
```

Comment:
using the cause seems like pretty delicate logic that could be broken by changes to RepositoryDelegatorFunction.